api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -61,6 +61,13 @@ public void validate(Object target, Errors errors) {
 		ValidationUtils.rejectIfEmpty(errors, "patient", "allergyapi.patient.required");
 		
 		Allergy allergy = (Allergy) target;
+
+		if(allergy.getReactionNonCoded() != null) {
+			if(!allergy.getReactionNonCoded().matches("[a-zA-Z]+$")) {
+				errors.rejectValue("reactionNonCoded", "Other Reaction must not contain a number or symbol");
+			}
+			}
```

Comment:
Collect the identation here following  guidelines[ here](https://wiki.openmrs.org/display/docs/Developer+How-To+Setup+And+Use+Eclipse)